napari/layers/shapes/_accelerated_triangulate.py

```diff
@@ -319,11 +342,139 @@ def _orientation(p1: np.ndarray, p2: np.ndarray, p3: np.ndarray) -> float:
+            - start_angle
+        )
+        if angle < 0:
+            angle += 2 * pi
+        if angle < prev_angle:
+            return False
+        prev_angle = angle
+    return True
+
+
+@njit(cache=True, inline='always')
+def _centroid(vertices: CoordinateArray2D) -> np.ndarray:
+    """Calculate the centroid of a polygon.
+
+    Parameters
+    ----------
+    vertices : np.ndarray
+        Array of vertex coordinates with shape (N, 2)
+
+    Returns
+    -------
+    np.ndarray
+        The centroid of the polygon
+    """
+    vertices_sum = np.zeros(2, dtype=np.float32)
+    for i in range(len(vertices)):
+        vertices_sum[0] += vertices[i][0]
+        vertices_sum[1] += vertices[i][1]
+    return vertices_sum / len(vertices)
+
+
+@njit(cache=True, inline='always')
+def is_convex(vertices: CoordinateArray2D) -> bool:
+    """Check if a polygon is convex.
+
+    Parameters
+    ----------
+    vertices : np.ndarray
+        Array of vertex coordinates with shape (N, 2)
+
+    Returns
+    -------
+    bool
+        True if the polygon is convex, False otherwise
+    """
+    # A polygon is convex if all its internal angles are less than 180 degrees.
+    # This is equivalent to checking that the orientation of all consecutive
+    # triples of points is the same.
+    if len(vertices) < 3:
+        return False
+    if len(vertices) == 3:
+        return True
+    orientation_ = Orientation.collinear
+    idx = 0
+    while idx < vertices.shape[0] - 2:
+        p1 = vertices[idx]
+        p2 = vertices[idx + 1]
+        p3 = vertices[idx + 2]
+        idx += 1
+        triangle_orientation = _orientation(p1, p2, p3)
+        if triangle_orientation != Orientation.collinear:
+            orientation_ = triangle_orientation
+            break
+    if orientation_ == Orientation.collinear:
+        return False
+    while idx < vertices.shape[0] - 2:
+        p1 = vertices[idx]
+        p2 = vertices[idx + 1]
+        p3 = vertices[idx + 2]
+        idx += 1
+        triangle_orientation = _orientation(p1, p2, p3)
+        if (
+            triangle_orientation != Orientation.collinear
+            and triangle_orientation != orientation_
+        ):
+            return False
+
+    triangle_orientation = _orientation(
+        vertices[idx], vertices[idx + 1], vertices[0]
     )
-    # fmt: on
+    if (
+        triangle_orientation != Orientation.collinear
+        and triangle_orientation != orientation_
+    ):
+        return False
+
+    triangle_orientation = _orientation(
+        vertices[idx + 1], vertices[0], vertices[1]
+    )
+    if (
+        triangle_orientation != Orientation.collinear
+        and triangle_orientation != orientation_
+    ):
+        return False
+
+    centroid = _centroid(vertices)
+
+    if orientation_ == Orientation.anticlockwise:
+        return _are_polar_angles_monotonic(vertices, centroid)
+
+    return _are_polar_angles_monotonic(vertices[::-1], centroid)
```

Comment:
```suggestion
    # all the angles had the same orientation. We finally check that
    # there are no self intersections by checking the angles the
    # vertices make with the centroid.
    centroid = _centroid(vertices)

    if orientation_ == Orientation.anticlockwise:
        return _are_polar_angles_monotonic(vertices, centroid)

    return _are_polar_angles_monotonic(vertices[::-1], centroid)
```

Though tbh I would consider moving the orientation check inside the monotonic function:

```python
    # all the angles had the same orientation. We finally check that
    # there are no self intersections by checking the angles the
    # vertices make with the centroid.
    centroid = _centroid(vertices)

    return _are_polar_angles_monotonic(vertices, centroid, current_orientation)
```

Reply:
Adding `current_orientation` to `_are_polar_angles_monotonic` is reducing readability for me.